server/graphql/v2/object/Host.js

```diff
@@ -437,6 +464,120 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          account: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          dateFrom: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics beginning from this date.',
+          },
+          dateTo: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics until this date.',
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = {
+            HostCollectiveId: host.id,
+            kind: TransactionKind.CONTRIBUTION,
+            type: TransactionTypes.CREDIT,
+          };
+          const numberOfDays = getNumberOfDays(args.dateFrom, args.dateTo, host);
+          const dateRange = getFilterDateRange(args.dateFrom, args.dateTo);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          if (args.account) {
+            const collectives = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
+            const collectiveIds = collectives.map(collective => collective.id);
+            where.CollectiveId = { [Op.in]: collectiveIds };
+          }
+          const contributionsCount = await models.Transaction.count({
+            where,
+          });
+          const contributionsAmountSum = await models.Transaction.sum('amount', { where });
+          const dailyAverageIncomeAmount = contributionsAmountSum ? contributionsAmountSum / numberOfDays : 0;
+          return {
+            contributionsCount,
+            oneTimeContributionsCount: models.Transaction.count({
+              where,
+              include: [{ model: models.Order, where: { interval: null } }],
+            }),
+            recurringContributionsCount: models.Transaction.count({
+              where,
+              include: [{ model: models.Order, where: { interval: { [Op.ne]: null } } }],
```

Comment:
I think this one will return erroneous values: if you have one monthly recurring contribution with 8 transactions, it will return `recurringContributionsCount=8`. We should group by `OrderId`. Same with `oneTimeContributionsCount`, though we're not supposed to have multiple transactions for the same order for this one, but just in case...

Reply:
Thanks for the catch. I've corrected it for both `recurringContributionsCount` and `oneTimeContributionsCount`. 👍🏽